air-quality-ui/system_tests/ui/banner/banner.spec.ts

```diff
@@ -0,0 +1,142 @@
+import { expect, test } from '../../utils/fixtures'
+import { gotoPage, setupPageWithMockData } from '../../utils/helper_methods'
+import { createForecastAPIResponseData } from '../../utils/mocked_api'
+;[
+  {
+    url: '/city/Rio%20de%20Janeiro',
+    pageType: 'city',
+  },
+  {
+    url: '/city/summary',
+    pageType: 'summary',
+  },
+].forEach(({ url, pageType }) => {
+  test.describe('Basic checks', () => {
+    test.beforeEach(async ({ page }) => {
+      await gotoPage(page, url)
+    }),
+      test(`vAirify logo is correct on ${pageType} page`, async ({
+        banner,
+      }) => {
+        await banner.logo.scrollIntoViewIfNeeded()
+        const logoShot = await banner.logo.screenshot()
+        expect(logoShot).toMatchSnapshot('vAirify-logo.png')
+      }),
+      test(`Date picker is visible on ${pageType} page`, async ({ banner }) => {
+        await expect(banner.datePicker).toBeVisible()
+      })
+    test(`Verify date picker times can only be 12:00 or 00:00 on ${pageType} page`, async ({
+      banner,
+    }) => {
+      await banner.calendarIcon.click()
+
+      await expect(banner.datePickerTimeOptions).toHaveCount(2)
+      await expect(banner.datePickerTimeOption0000).toBeVisible()
+      await expect(banner.datePickerTimeOption1200).toBeVisible()
+    })
+  })
+})
+;[
+  {
+    url: '/city/Rio%20de%20Janeiro',
+    pageType: 'city',
+  },
+  {
+    url: '/city/summary',
+    pageType: 'summary',
+  },
+].forEach(({ url, pageType }) => {
+  test.describe('Future dates and times are not possible', () => {
+    test.beforeEach(async ({ page, banner }) => {
+      const mockSystemDate: Date = new Date('2024-07-26T10:00:00Z')
+
+      await page.clock.setFixedTime(mockSystemDate)
+      await gotoPage(page, url)
+      await banner.calendarIcon.click()
+    }),
+      test(`Date picker cannot select a future day on ${pageType} page`, async ({
+        banner,
+      }) => {
+        await expect(banner.futureDay27).toBeDisabled()
+
+        await banner.datePickerYearOpenButton.click()
```

Comment:
You've repeated the check for 2025 a few times, and have a specific test for it so no need to reproduce each time. Let's just check it once

Reply:
fixed